internal/app/summon/command/temp_factory_test.go

```diff
@@ -0,0 +1,88 @@
+package command
+
+import (
+	"io/ioutil"
+	"os"
+	"strings"
+	"testing"
+
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+// TODO: Test TempFactory.Cleanup()
+// TODO: Test TempFactory.Push()
+
+func splitEq(s string) (string, string) {
+	a := strings.SplitN(s, "=", 2)
+	return a[0], a[1]
+}
+
+type envSnapshot struct {
+	env []string
+}
+
+func ClearEnv() *envSnapshot {
+	e := os.Environ()
+
+	for _, s := range e {
+		k, _ := splitEq(s)
+		os.Setenv(k, "")
+	}
+	return &envSnapshot{env: e}
+}
+
+func (e *envSnapshot) RestoreEnv() {
+	ClearEnv()
+	for _, s := range e.env {
+		k, v := splitEq(s)
+		os.Setenv(k, v)
+	}
+}
+
```

Comment:
Just a thought, not sure if possible, but seems like it would be a lot cleaner to just execute the tests in a subshell rather snapshotting and restoring the env.  Could that be done somehow?

Reply:
I haven't researched it but given that it would require a whole subshell-launching harness for running and collecting the results which would also be pretty fragile, I don't think we should try to do that as part of this PR. We also use this exact same setup [elsewhere](https://github.com/cyberark/conjur-api-go/blob/f60f81f91778842e743b098f84210ba7d5f5d637/conjurapi/config_test.go#L120-L121).